api/app.go

```diff
@@ -0,0 +1,295 @@
+// Copyright (c) 2015 Spinpunch, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+package api
+
+import (
+	l4g "code.google.com/p/log4go"
+	"fmt"
+	"github.com/gorilla/mux"
+	"github.com/mattermost/platform/model"
+	"github.com/mattermost/platform/utils"
+	"net/http"
+	"net/url"
+)
+
+func InitApp(r *mux.Router) {
+	l4g.Debug("Initializing app api routes")
+
+	sr := r.PathPrefix("/apps").Subrouter()
+
+	sr.Handle("/oauth2/register", ApiUserRequired(registerOAuthApp)).Methods("POST")
```

Comment:
especially when we start talking about app center apps, and admin console, which would be the root app.